src/lib/statistics/generate_pruning_statistics.cpp

```diff
@@ -86,15 +92,21 @@ void generate_chunk_pruning_statistics(const std::shared_ptr<Chunk>& chunk) {
 
 void generate_chunk_pruning_statistics(const std::shared_ptr<Table>& table) {
   const auto chunk_count = table->chunk_count();
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
+
   for (auto chunk_id = ChunkID{0}; chunk_id < chunk_count; ++chunk_id) {
-    const auto chunk = table->get_chunk(chunk_id);
+    const auto chunk = table->get_chunk(ChunkID{chunk_id});
 
     if (!chunk || chunk->is_mutable()) {
-      continue;
+      return;
```

Comment:
I think this should stay `continue`. Otherwise, we don't do anything if the table's final chunk is mutable or there is a physically deleted chunk. Remember that this is also called when adding a table to the storage manager, not only during initial loading.

Reply:
You're absolutely right, stupid me forgot to change it back when moving it out of the lambda. Oops :see_no_evil: